geps/gep-1494/index.md

```diff
@@ -130,20 +130,386 @@ From @ongy, some additional goals to keep in mind:
+and configuration of a HTTP based API (which, as it's not defined using a
+specification like protobuf, requires more configuration).
+
+The important thing to remember here is that the actual authorization process
+is delegated to the authorization service, and the authentication process _may
+optionally_ also be delegated there - which is why the ext_authz approach allows
+handling many Auth methods - most of the work is performed by external services
+which implement various methods (like Basic Auth, OAuth, JWT validation, etc).
+
+#### Why use it over other options?
+
+The community discussed Auth extensively in-person at Kubecon London in early 2025,
+and got broad agreement from multiple dataplanes that:
+
+* something like ext_authz was a good idea, because it's flexible and allows the
+  implementation of many types of Auth without specific protocol implementation
+  in upstream
+* Envoy's ext_authz protocol has no major problems that would stop us using it
+* Envoy-based implementations mostly already have support for it
+
+At that meeting, those present agreed that ext_authz was a good place to start.
+
+Most non-Envoy dataplanes also already have similar methods, so the maintainers
+of projects using other dataplanes were okay with this idea.
+
+The alternative here would be to add a Filter type _per auth method_, which, given
+the large number of options, could quickly become very complex.
+
+This GEP is, however, explicitly _not_ ruling out the possibility of adding
+specific Filters for specific Auth methods in the future, if users of this API
+find the overhead of running a compatible implementation to be too much.
+
+### API Design
+
+#### Phase 1: Adding a Filter
+
+This config mainly takes inspiration from Envoy's ext_authz filter config, while
+also trying to maintain compatibility with other HTTP methods.
+
+This design is also trying to start with a minimum feature set, and add things
+as required, rather than adding everything configurable in all implementations
+immediately.
+
+There is some difference between data planes, based on the links above, but
+these fields should be broadly supportable across all the listed implementations.
+
+Some design comments are included inline.
+
+The intent for Phase 2 is that this struct will be included in an eventual Policy
+so that additions only need to be made in one place.
+
+Additionally, as per other added Filters, the config is included in HTTPRoute,
+and is not an additional CRD.
+
+##### Go Structs
+
+```go
+
+// HTTPRouteExtAuthProtcol specifies what protocol should be used
+// for communicating with an external authorization server.
+//
+// Valid values are supplied as constants below.
+type HTTPRouteExtAuthProtocol string
+
+const (
+	HTTPRouteExtAuthGRPCProtocol HTTPRouteExtAuthProtocol = "GRPC"
+	HTTPRouteExtAuthHTTPProtocol HTTPRouteExtAuthProtocol = "HTTP"
+)
+// HTTPExtAuthFilter defines a filter that modifies requests by sending
```

Comment:
How do we get consistent interop with features like frontend mTLS? Ideally we can present the frontend mTLS extracted identity to the ext_authz server. The ext_authz protocol has a notion of attribute context, will this be required as part of compliance across proxy implementations? I thought I had opened a thread on this but don't see it.